source/common/stats/thread_local_store.cc

```diff
@@ -594,7 +594,12 @@ Gauge& ThreadLocalStoreImpl::ScopeImpl::gaugeFromStatNameWithTags(
   TagUtility::TagStatNameJoiner joiner(prefix_.statName(), name, stat_name_tags, symbolTable());
   StatName final_stat_name = joiner.nameWithTags();
 
-  StatsMatcher::FastResult fast_reject_result = parent_.fastRejects(final_stat_name);
+  StatsMatcher::FastResult fast_reject_result;
+  if(import_mode != Gauge::ImportMode::Hidden) {
```

Comment:
a minor point but we should leave a TODO to rename ImportMode go be something more generic. Not sure what name I like best but it's not just import that's being controlled anymore.

Maybe just Mode (which is already inside Gauge namespace) so it'll read like Gauge::Mode::Accumulate or Gauge::Mode::NeverImport or Gauge::Mode::Hidden, which seems clear enough and not redundant or verbose.

Reply:
I am not asking you to do it in this PR -- I think big refactors touching hundreds of lines of unrelated files with a simple rename are easier to do as a separate PR that changes zero semantics

Reply:
Done. I added a few TODOs for myself at different spots